spinalcordtoolbox/scripts/sct_register_to_template.py

```diff
@@ -592,13 +592,22 @@ def main(argv: Sequence[str]):
 
             # Apply straightening to labels
             printv('\nApply straightening to labels...', verbose)
+            label_vals_src = {coord.value for coord in Image(ftmp_label).getCoordinatesAveragedByValue()}
             sct_apply_transfo.main(argv=[
                 '-i', ftmp_label,
                 '-o', add_suffix(ftmp_label, '_straight'),
                 '-d', add_suffix(ftmp_seg, '_straight'),
                 '-w', 'warp_curve2straight.nii.gz',
                 '-x', 'nn'])
             ftmp_label = add_suffix(ftmp_label, '_straight')
+            label_vals_out = {coord.value for coord in Image(ftmp_label).getCoordinatesAveragedByValue()}
+            missing_labels = label_vals_src - label_vals_out
+            if missing_labels:
+                printv(
+                    f"ERROR: Labels {missing_labels} were lost during straightening transform. This can be caused by "
+                    f"the labels being outside the ROI of the spinal cord segmentation. Please make sure all labels "
+                    f"are within the ROI of the spinal cord segmentation.", type='error'
+                )
 
```

Comment:
In https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/3873#issue-1363434092, I proposed 2 possible checks:


>  * For example, we could check the coordinates of the nonzero labels, then check the dimensions of the warping field, then throw a warning/exception if we can determine that that the label won't be transformed as part of the straightening.
>  * Or, more simply, we could compare the # of labels before straightening to the # of labels after straightening, and emit a warning/error if a label was lost in the straightening.


I was hesitant to go for the former option (checking the coordinates of the label relative to the warping field) because of the dilation step; there's a possible case where the single-voxel label is outside of the warping field, but the dilation step expands the label to be inside the warping field. So, I wasn't sure: should I check against the undilated label or the dilated label? Neither guarantees for sure that the label will be lost/kept.

So, that's why I felt that checking the presence of labels post-straightening was more foolproof.